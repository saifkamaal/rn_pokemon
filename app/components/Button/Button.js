// @flow
import React, { Component } from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";
export interface Props {
  onPress: Function;
  text: string;
}
export interface State {}
class Button extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <TouchableOpacity style={styles.buttonStyle} onPress={this.props.onPress}>
        <Text style={styles.textStyle}>{this.props.text}</Text>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  buttonStyle: {
    alignSelf: "stretch",
    backgroundColor: "#fff",
    borderRadius: 5,
    borderWidth: 1,
    borderColor: "#007aff",
    marginHorizontal: 5
  },
  textStyle: {
    alignSelf: "center",
    color: "#007aff",
    fontSize: 16,
    fontWeight: "600",
    marginVertical: 10,
    marginHorizontal: 10
  }
});

export default Button;
