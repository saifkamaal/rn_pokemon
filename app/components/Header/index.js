import React, { Component } from "react";
import { Text, View, StyleSheet, Button } from "react-native";

class Header extends Component {
  onBackPress = () => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  render() {
    const { showBack, title } = this.props;
    return (
      <View style={styles.header}>
        <View style={styles.button}>
          {showBack && (
            <Button color={"#007aff"} onPress={this.onBackPress} title="Back" />
          )}
        </View>
        <Text style={styles.text}>{title}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  header: {
    marginTop: 20,
    height: 60,
    backgroundColor: "transparent",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between"
  },
  text: {
    textAlign: "center",
    color: "black",
    paddingRight: 10,
    flexGrow: 1,
    fontSize: 16
  },
  button: {
    position: "absolute",
    zIndex: 2
  }
});

export default Header;
