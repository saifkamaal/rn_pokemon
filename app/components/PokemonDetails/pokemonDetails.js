// @flow
import React, { Component } from "react";
import { View, Text, Image, StyleSheet } from "react-native";

export interface Props {
  pokemonDetails: Object;
}
export interface State {
  name: string;
  id: string;
  weight: string;
  abilities: Object;
  moves: Object;
  images: Object;
  stats: Object;
}
class PokemonDetails extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      id: "",
      images: {},
      weight: "",
      name: "",
      abilities: [],
      moves: [],
      stats: []
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    if (this.props.pokemonDetails != nextProps.pokemonDetails) {
      const pokemonDetails = nextProps.pokemonDetails;
      this.setState({
        id: pokemonDetails.id,
        images: pokemonDetails.sprites,
        weight: pokemonDetails.weight,
        name: pokemonDetails.name,
        abilities: pokemonDetails.abilities,
        moves: pokemonDetails.moves,
        stats: pokemonDetails.stats
      });
    }
  }

  render() {
    const id = this.state.id;
    const name = this.state.name;
    const weight = this.state.weight;
    const abilities = this.state.abilities;
    const moves = this.state.moves;
    const image = this.state.images;
    const stats = this.state.stats;
    return (
      <View style={{ backgroundColor: "white" }}>
        <View style={styles.imageContainer}>
          <Image
            source={{
              uri: image.front_default || "https://i.imgur.com/es7sovA.jpg"
            }}
            style={styles.image}
          />
          <Text style={styles.name}>
            {name.charAt(0).toUpperCase() + name.slice(1)}
          </Text>
        </View>
        <View style={{ margin: 20 }}>
          <Text style={{ fontSize: 28 }}>Details</Text>
          <View
            style={{
              flexDirection: "row",
              justifyContent: "space-around",
              margin: 20
            }}
          >
            <View>
              <Text style={{ fontSize: 20 }}>WEIGHT:</Text>
              <Text style={{ paddingLeft: 20 }}>{weight}</Text>
              <Text style={{ fontSize: 20, marginTop: 20 }}>ABILITIES:</Text>
              {abilities.map((a, i) => (
                <View key={i}>
                  <Text style={{ paddingLeft: 20 }}>{a.ability.name}</Text>
                </View>
              ))}
              <Text style={{ fontSize: 20, marginTop: 20 }}>STATS:</Text>
              {stats.map((a, i) => (
                <View key={i}>
                  <Text style={{ paddingLeft: 20 }}> {a.stat.name}</Text>
                </View>
              ))}
            </View>
            <View>
              <Text style={{ fontSize: 20 }}>MOVES:</Text>
              <View style={{ flexDirection: "column" }}>
                {moves.map((a, i) => (
                  <Text key={i} style={{ paddingLeft: 20 }}>
                    {a.move.name}
                  </Text>
                ))}
              </View>
            </View>
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  image: {
    height: 200,
    width: 200,
    flex: 1
  },
  imageContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-around",
    backgroundColor: "#007aff"
  },
  name: {
    color: "#fff",
    fontSize: 33,
    marginTop: 80,
    flex: 1
  }
});

export default PokemonDetails;
