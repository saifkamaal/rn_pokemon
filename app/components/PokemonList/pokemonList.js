import React, { Component } from "react";
import { List, ListItem } from "react-native-elements";

export interface Props {
  pokemons: Object;
  onPress: Function;
}
export interface State {}
class PokemonList extends Component<Props, State> {
  constructor() {
    super();
  }

  render() {
    return (
      <List>
        {this.props.pokemons.map((pokemon, i) => (
          <ListItem
            button
            onPress={() => this.props.onPress(pokemon.url)}
            key={i}
            title={pokemon.name.charAt(0).toUpperCase() + pokemon.name.slice(1)}
          />
        ))}
      </List>
    );
  }
}

export default PokemonList;
