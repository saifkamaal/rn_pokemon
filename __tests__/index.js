import React from "react";
import Enzyme, { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Button from "../app/components/Button/Button";
import Header from "../app/components/Header";
import PokemonList from "../app/components/PokemonList/pokemonList";
import PokemonDetails from "../app/components/PokemonDetails/pokemonDetails";

Enzyme.configure({ adapter: new Adapter() });

describe("Testing Button component", () => {
  it("renders as expected", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Testing Header component", () => {
  it("renders as expected", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Testing PokemonDetails component", () => {
  it("renders as expected", () => {
    const wrapper = shallow(<PokemonDetails />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Testing PokemonList component", () => {
  it("renders as expected", () => {
    const pokemons = [
      {
        name: "Saif",
        url: "http://google.com"
      }
    ];
    const wrapper = shallow(<PokemonList pokemons={pokemons} />);
    expect(wrapper).toMatchSnapshot();
  });
});
