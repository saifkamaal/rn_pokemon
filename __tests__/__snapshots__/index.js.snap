// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component renders as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Header component renders as expected 1`] = `ShallowWrapper {}`;

exports[`Testing PokemonDetails component renders as expected 1`] = `ShallowWrapper {}`;

exports[`Testing PokemonList component renders as expected 1`] = `ShallowWrapper {}`;
